packages/vaadin-notification/src/vaadin-notification.js

```diff
@@ -315,88 +297,50 @@ class NotificationElement extends ThemePropertyMixin(ElementMixin(PolymerElement
        * - `notification` The reference to the `<vaadin-notification>` element.
        * @type {!NotificationRenderer | undefined}
        */
-      renderer: Function,
-
-      /**
-       * The template of the notification card content.
-       * @type {!HTMLTemplateElement | undefined}
-       * @protected
-       */
-      _notificationTemplate: Object
+      renderer: Function
     };
   }
 
   static get observers() {
-    return [
-      '_durationChanged(duration, opened)',
-      '_templateOrRendererChanged(_notificationTemplate, renderer, opened)'
-    ];
+    return ['_durationChanged(duration, opened)', '_rendererChanged(renderer, opened)'];
   }
 
   /** @protected */
   ready() {
     super.ready();
 
-    this._observer = new FlattenedNodesObserver(this, (info) => {
-      this._setTemplateFromNodes(info.addedNodes);
-    });
-  }
+    this._card = this.$['vaadin-notification-card'];
 
-  /**
-   * @param {!Array<!Node>} nodes
-   * @protected
-   */
-  _setTemplateFromNodes(nodes) {
-    this._notificationTemplate =
-      nodes.find((node) => node.localName && node.localName === 'template') || this._notificationTemplate;
+    if (window.Vaadin && window.Vaadin.templateRendererCallback) {
+      window.Vaadin.templateRendererCallback(this);
+    }
   }
 
   /**
    * Manually invoke existing renderer.
    */
   render() {
-    if (typeof this.renderer !== 'function') {
-      return;
-    }
-    this.renderer(this._card, this);
-  }
+    if (!this.renderer) return;
 
-  /** @private */
-  _removeNewRendererOrTemplate(template, oldTemplate, renderer, oldRenderer) {
-    if (template !== oldTemplate) {
-      this._notificationTemplate = undefined;
-    } else if (renderer !== oldRenderer) {
-      this.renderer = undefined;
-    }
+    this.renderer(this._card, this);
+    this._card.setAttribute('aria-label', this._card.textContent.trim());
```

Comment:
Originally A11y tests expected this attribute only when using Polymer templates. Since I had passed Polymer templates as a renderer function, I had to add this attribute here to make tests pass.

Probably, I should have removed this attribute at all. What do you think?

Reply:
Seems that the overlay content gets properly announced (VoiceOver, JAWS, NVDA) when the notification opens if the"aria-label" is not added.
If the separate "aria-label" is there, seems that the notification content gets announced twice on JAWS and NVDA. I'd vote for removing the attribute altogether.

Reply:
Thanks for in-depth look! I will remove this attribute then.

Reply:
Removed.